### PR TITLE
Add debug view, embed pdb

### PR DESCRIPTION
**Pull request by warquys**
When I use debug mode, there is very quickly too much information that can be compressed onto the summary when I inspect an item.
I reduced the important properties to 4 and named them to make things clearer.
I also avoided invoking the sub ToString by only displaying the most important prop. In the case of Rooms.

the reasons why the pdb is added directly to the assembly, this will make distribution easier, I have seen too many logs which do not include the line causing the error in Exiled. This edition. This edition will make it easier to read errors for new devs. and will facilitate debugging without having to recompile exiled.

Changes made to Attachment Identifier make code easier to read and faster.
I'll let you judge it for yourself.

### CODE REVIEW

File: Exiled.API/Features/GlobalPatchProcessor.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ namespace Exiled.API.Features
     /// <summary>
     /// A <see cref="Harmony"/> alternative detour tool which adds more ways to manage patches and external assemblies.
     /// </summary>
-    public class GlobalPatchProcessor
+    public static class GlobalPatchProcessor
```
Comment (louis1706) [inline Exiled.API/Features/GlobalPatchProcessor.cs:23]: There a breaking change

**Review by louis1706 — CHANGES_REQUESTED** [review]

**Comment by louis1706** [comment]
Also i feel like this should be kept inside .ToString() but i will let Nao handle your PR

File: Exiled.API/Features/DamageHandlers/AttackerDamageHandler.cs [hunk 2]
```diff
@@ -115,5 +118,15 @@ public override void ProcessDamage(Player player)
                 Damage = 0f;
             }
         }
+
+        private string GetDebuggerDisplay()
+        {
+            var debugView = $"DamageHandler Damage = {Damage}";
```
Comment (NaoUnderscore) [inline Exiled.API/Features/DamageHandlers/AttackerDamageHandler.cs:124]: No `var` allowed.
Same goes for all occurrences.

File: Exiled.API/Structs/AttachmentIdentifier.cs [hunk 3]
```diff
@@ -138,14 +140,14 @@ internal AttachmentIdentifier(uint code, AttachmentName name, AttachmentSlot slo
         /// <returns><see langword="true"/> if <see cref="string"/> was converted successfully; otherwise, <see langword="false"/>.</returns>
         public static bool TryParse(string s, out AttachmentIdentifier identifier)
         {
-            identifier = default;
-
-            foreach (AttachmentIdentifier attId in Features.Items.Firearm.AvailableAttachments.Values.SelectMany(kvp => kvp.Where(kvp2 => kvp2.Name.ToString() == s)))
+            foreach(var attIds in Features.Items.Firearm.AvailableAttachments.Values)
```
Comment (NaoUnderscore) [inline Exiled.API/Structs/AttachmentIdentifier.cs:143]: Don't really get it.
As long as it works and it has been tested, I guess it's fine.
I need your feedback though.

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

**Comment by NaoUnderscore** [comment]
Also, you may consider to use properties instead of `Get` methods.

**Comment by VALERA771** [comment]
I can't get a point of adding `DebuggerDisplay`. Debugging plugins for unity projects is not available in IDEs

File: Exiled.API/Features/DamageHandlers/AttackerDamageHandler.cs (hunk 2 above)
Comment (warquys) [reply Exiled.API/Features/DamageHandlers/AttackerDamageHandler.cs:124]: I just want to warn you that I found this line [Consuming.cs](https://github.com/Exiled-Team/EXILED/blob/84e496ffb9cc7c45c27b10c86a6989c88faad7ea/Exiled.Events/Patches/Events/Scp0492/Consuming.cs#L68) which is not mine but which contains a "var" when I was replacing the "var"s

File: Exiled.API/Structs/AttachmentIdentifier.cs (hunk 3 above)
Comment (warquys) [reply Exiled.API/Structs/AttachmentIdentifier.cs]: I would make these changes in another push and explain them more clearly.

**Comment by warquys** [comment]
All the GetDebugDisplay are change to prop and rename to DebugDiplay.

**Comment by warquys** [comment]
> I can't get a point of adding `DebuggerDisplay`. Debugging plugins for unity projects is not available in IDEs

I use Visual Studio and is valid for unity debugging.

File: Exiled.API/Features/GlobalPatchProcessor.cs (hunk 1 above)
Comment (warquys) [reply Exiled.API/Features/GlobalPatchProcessor.cs]: back to normal

File: Exiled.API/Features/Roles/OverwatchRole.cs [hunk 4]
```diff
@@ -7,13 +7,16 @@
...
+    using System.Diagnostics;
+
     using PlayerRoles;
 
     using OverwatchGameRole = PlayerRoles.Spectating.OverwatchRole;
 
     /// <summary>
     /// Defines a role that represents a player with overwatch enabled.
     /// </summary>
+    [DebuggerDisplay("Overwat")]
```
Comment (NaoUnderscore) [inline Exiled.API/Features/Roles/OverwatchRole.cs:19]:
```suggestion
    [DebuggerDisplay("Overwatch")]
```

File: Exiled.API/Features/Room.cs [hunk 5]
```diff
@@ -27,6 +28,7 @@ namespace Exiled.API.Features
     /// <summary>
     /// The in-game room.
     /// </summary>
+    [DebuggerDisplay("Tpe = {Type} Name = {Name}")]
```
Comment (NaoUnderscore) [inline Exiled.API/Features/Room.cs:31]:
```suggestion
    [DebuggerDisplay("Type = {Type} Name = {Name}")]
```

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

**Comment by ALEXWARELLC** [comment]
@NaoUnderscore Consider looking into merging or closing this PR.

File: Exiled.API/Features/Camera.cs [hunk 6]
```diff
@@ -210,6 +212,9 @@ public bool IsBeingUsed
             set => Base.IsActive = value;
         }
 
+        [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+        private string DebuggerDisplay => $"Room = {Room?.RoomName.ToString() ?? "Unknow"} Name = {Name} Id = {Id} Used = {IsBeingUsed}";
```
Comment (ALEXWARELLC) [inline Exiled.API/Features/Camera.cs:216]:
```suggestion
        private string DebuggerDisplay => $"Room = {Room?.RoomName.ToString() ?? "Unknown"} Name = {Name} Id = {Id} Used = {IsBeingUsed}";
```

File: Exiled.API/Features/Doors/Door.cs [hunk 7]
```diff
@@ -288,6 +290,9 @@ public Vector3 Scale
         /// </summary>
         internal List<Room> RoomsValue { get; } = new List<Room>();
 
+        [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+        private string DebuggerDisplay => $"Room = {Room?.RoomName.ToString() ?? "Unknow"} Type = {Type} Open = {IsOpen}";
```
Comment (ALEXWARELLC) [inline Exiled.API/Features/Doors/Door.cs:294]:
```suggestion
        private string DebuggerDisplay => $"Room = {Room?.RoomName.ToString() ?? "Unknown"} Type = {Type} Open = {IsOpen}";
```

File: Exiled.API/Features/Generator.cs [hunk 8]
```diff
@@ -226,6 +228,9 @@ public KeycardPermissions KeycardPermissions
             set => Base._requiredPermission = (Interactables.Interobjects.DoorUtils.KeycardPermissions)value;
         }
 
+        [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+        private string DebuggerDisplay => $"Room = {Room?.RoomName.ToString() ?? "Unknow"} State = {State} KeycardPermissions = {KeycardPermissions}";
```
Comment (ALEXWARELLC) [inline Exiled.API/Features/Generator.cs:232]:
```suggestion
        private string DebuggerDisplay => $"Room = {Room?.RoomName.ToString() ?? "Unknown"} State = {State} KeycardPermissions = {KeycardPermissions}";
```

File: Exiled.API/Features/Items/MicroHid.cs [hunk 9]
```diff
@@ -7,13 +7,16 @@
...
+    using System.Diagnostics;
+
     using Exiled.API.Features.Core;
     using Exiled.API.Interfaces;
     using InventorySystem.Items.MicroHID;
 
     /// <summary>
     /// A wrapper class for <see cref="MicroHIDItem"/>.
     /// </summary>
+    [DebuggerDisplay("MicroHid")]
```
Comment (ALEXWARELLC) [inline Exiled.API/Features/Items/MicroHid.cs:19]:
```suggestion
    [DebuggerDisplay("MicroHID")]
```

File: Exiled.API/Features/Broadcast.cs [hunk 10]
```diff
@@ -8,6 +8,9 @@
 namespace Exiled.API.Features
 {
     using System.ComponentModel;
+    using System.Diagnostics;
+
+    using YamlDotNet.Serialization;
```
Comment (ALEXWARELLC) [inline Exiled.API/Features/Broadcast.cs:13]: Please fix the CI Failure located here.

**Review by ALEXWARELLC — CHANGES_REQUESTED** [review]

**Comment by ALEXWARELLC** [comment]
@warquys The main concern I have is that while this is good for development environments where most plugins will be developed, and debugged, but it can become detrimental to space and performance in high-performance applications like a game server.

If you can make a configurable option for this, that would make this even better for different environments.

**Comment by ALEXWARELLC** [comment]
@NaoUnderscore Consider closing this PR for being stale.

**Comment by warquys** [comment]
> @warquys The main concern I have is that while this is good for development environments where most plugins will be developed, and debugged, but it can become detrimental to space and performance in high-performance applications like a game server.
> 
> If you can make a configurable option for this, that would make this even better for different environments.

Create a build condition for Developer and a new version of Exiled.
Expect space, I don't see any performance issues.

**Pull request closed without merging**